win32ss/gdi/ntgdi/palette.c

```diff
@@ -1247,7 +1247,7 @@ NtGdiUnrealizeObject(HGDIOBJ hgdiobj)
    PPALETTE palGDI;
 
    if ( !hgdiobj ||
-        ((UINT)hgdiobj & GDI_HANDLE_STOCK_MASK) ||
+        ((UINT_PTR)hgdiobj & GDI_HANDLE_STOCK_MASK) ||
```

Comment:
HandleToUlong ?

Reply:
There is not reason to truncate the value. If at some point the handle mask is changed to bit 48, then nobody would notice this truncation. Also in the reverse case (UlongToHandle), the result is wrongly sign extended, so this function is entirely wrong, and therefore both should not be used with GDI handles.